fiftyone/brain/internal/core/leaky_splits.py

```diff
@@ -115,312 +147,241 @@ def method(self):
-        if self._last_computed_threshold == self._leak_threshold:
-            return self._leaks
-
-        # populate index if it doesn't have some samples
-        if not self._similarity_index.total_index_size == len(self.samples):
-            (
-                embeddings,
-                sample_ids,
-                label_ids,
-            ) = self._similarity_index.compute_embeddings(self.samples)
-            self._similarity_index.add_to_index(
-                embeddings, sample_ids, label_ids
-            )
-
-        # check if duplicates already computed in the index
-        if not self._similarity_index.thresh == self._leak_threshold:
-            self._similarity_index.find_duplicates(self._leak_threshold)
-        duplicates = self._similarity_index.duplicates_view()
-
-        # filter duplicates to just those with neighbors in different splits
-        to_keep = []
-        for (
-            sample_id,
-            neighbors,
-        ) in self._similarity_index.neighbors_map.items():
-            keep_sample = False
             sample_split = self._id2split.get(sample_id, None)
             if sample_split is None:
-                _throw_index_is_bigger_warning(sample_id)
-                continue  # sample is in index but not passed to leaky_splits
-            leaks = []
+                continue
+
             for n in neighbors:
-                neighbor_split = self._id2split.get(n[0], None)
+                neighbor_id = n[0]
+                neighbor_split = self._id2split.get(neighbor_id, None)
                 if neighbor_split is None:
-                    _throw_index_is_bigger_warning(n[0])
                     continue
-                if not (neighbor_split == sample_split):
-                    # at least one of the neighbors is from a different split
-                    # we keep this one
-                    keep_sample = True
-                    leaks.append(n[0])
 
-            if keep_sample:
-                to_keep.append(sample_id)
-                # remove all other samples because they are all from the same split
-                to_keep = to_keep + leaks
+                if neighbor_split != sample_split:
+                    _leak_ids.append(neighbor_id)
+
+            if _leak_ids:
+                leak_ids.append(sample_id)
+                leak_ids.extend(_leak_ids)
+
+        self._leak_ids = leak_ids
+
+    def leaks_view(self):
+        """Returns a view containg all potential leaks generated by the last
+        call to :meth:`find_leaks`.
 
-        duplicates = duplicates.select(to_keep)
+        Returns:
+            a :class:`fiftyone.core.view.DatasetView`
+        """
+        if self._thresh is None:
+            raise ValueError("You must first call `find_leaks()`")
 
-        # cache to avoid recomputation
-        self._last_computed_threshold = self._leak_threshold
-        self._leaks = duplicates
+        return self.samples.select(self._leak_ids, ordered=True)
```

Comment:
The previous implementation used:

```py
duplicates_view().select(to_keep)
```

which is not ideal because `duplicates_view()` internally performs a potentially much larger `select(..., ordered=True)` operation across all duplicates, not just specifically leaks.

Directly doing `select(self._leak_ids, ordered=True)` gets right to the point.